### PR TITLE
[External] Support GLOG project

### PR DESCRIPTION
**Pull request by kygx-legend**
Resolve issue #146.

Examples:
`make TestLogMsg`
`GLOG_v=5 GLOG_minloglevel=0 GLOG_log_dir=. ./TestLogMsg  #  to file`
`GLOG_v=5 GLOG_minloglevel=0 GLOG_logtostderr=1 ./TestLogMsg  #  to console`

### CODE REVIEW

**Comment by kygx-legend** [comment]
@ddmbr please review.

File: base/log.hpp [hunk 1]
```diff
@@ -16,6 +16,9 @@
 
 #include <string>
 
+// FIXME(Legend): This include is only used for test now, which is not self-efficient.
```
Comment (ddmbr) [inline base/log.hpp]: "self-sufficient"

Comment (kygx-legend) [inline base/log.hpp]: Oops

**Comment by kygx-legend** [comment]
@ddmbr 
1. As I tested, the configuration can be setup by the environment variable. But it can also be setup through boost::program_options, etc., log_dir.
2. When building the GLOG, the default switch of `enable multithreading` is open. Maybe we can test if this works.

**Comment by kygx-legend** [comment]
Some modification added. Please review again.

**Comment by ddmbr** [comment]
My patch clashed with yours

**Comment by kygx-legend** [comment]
Yes, I'm resolving that.

**Comment by kygx-legend** [comment]
Patch is rebased.

**Pull request merged**